### PR TITLE
More complete support for PEP440 version segments. (alpha, beta, ...)

### PR DESCRIPTION
**Pull request by jeduden**
Also, fixes version string when input is just a git tag.

### CODE REVIEW

File: pkgversion/pkgversion.py [hunk 1]
```diff
@@ -71,20 +80,54 @@ def pep440_version(version=get_version()):
...
-        v = re.compile(r'(\d+\.\d+(\.\d+)?)(-(\d+)-(\w+))?').search(version)
+        v = re.compile(r'\A(?P<githash>g\w+)\Z').match(version)
+        if v:
+            return '0.0+' + v.group('githash')
+
+        parts = []
+
+        head, match = match_tail(r'-(?P<gitnr>\d+)-(?P<githash>g\w+)', version)
+        if match:
+            parts.insert(0, '+' + match.group('gitnr') + '.' + match.group('githash'))
```
Comment (snelis) [inline pkgversion/pkgversion.py:103]: how about this:

``` python
parts.insert(0, '+{gitnr}.{githash}'.format(**match.groupdict()))
```

```
```

File: pkgversion/pkgversion.py [hunk 2]
```diff
@@ -71,20 +80,54 @@ def pep440_version(version=get_version()):
...
+
+        head, match = match_tail(r'.dev\d+', head)
+        if match:
+            parts.insert(0, match.group('tail'))
+
+        head, match = match_tail(r'.post\d+', head)
```
Comment (snelis) [inline pkgversion/pkgversion.py]: can we combine these as `.(dev|post)\d+`

Comment (snelis) [inline pkgversion/pkgversion.py]: better yet `\.(dev|post)\d+` escape the dot

Comment (jeduden) [inline pkgversion/pkgversion.py]: no unfortunately, not. both can be present.

File: pkgversion/pkgversion.py (hunk 1 above)
Comment (jeduden) [inline pkgversion/pkgversion.py:103]: nice ! fixed!

File: pkgversion/pkgversion.py (hunk 2 above)
Comment (jeduden) [inline pkgversion/pkgversion.py]: oh. but the dot needs to be escaped! you are right!

Comment (jeduden) [inline pkgversion/pkgversion.py]: fixed.

**Pull request merged**